### PR TITLE
Added Topological_Sort.js (Fixes #2159)

### LINKED ISSUE #2159: Topological Sort
- [x] C
- [x] C++
- [ ] C# - @himanshusharma89
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [x] Java
- [x] JS - @JanaSabuj
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by JanaSabuj**
Fixes #2159 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [ ] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Topological_Sort.js

#### Languages Used:
- JavaScript

#### Files Added:
- Topological_Sort.js 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by JanaSabuj** [comment]
@jainaman224 @asha15 @SakshiShreya please do review the PR.

**Comment by JanaSabuj** [comment]
please review the PR. @jainaman224 @asha15 @SakshiShreya

File: Topological_Sort/Topological_Sort.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            // show graph in readable format
+            console.log(`Vertex ${i} is connected to verices `, adj[i]);
+        }
+    };
+
+    addEdge = (x, y) => {
+        const {
+            nodes,
+            adj
+        } = this;
```
Comment (SakshiShreya) [inline Topological_Sort/Topological_Sort.js]: Just a suggestion. These lines could have been in one line.

File: Topological_Sort/Topological_Sort.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        visited[start] = true;
+
+        for (let x of adj[start]) {
+            if (!visited[x]) {
+                this.topoUtil(x, visited);
+            }
+        }
+
+        this.stack.push(start) // push the node when we have completed it
```
Comment (SakshiShreya) [inline Topological_Sort/Topological_Sort.js]: Add a semicolon. Maintain consistency.

File: Topological_Sort/Topological_Sort.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+// Program to implement Topological Sort
+class Graph {
+    constructor(nodes) {
+        //2 class variables
+        this.nodes = nodes; // no of vertices
+        this.adj = new Array(nodes); // adjacency list
+        for (let i = 0; i < nodes; i++) {
+            this.adj[i] = new Array();
+        }
+        this.stack = [] // a stack for storing the ending times of dfs
```
Comment (SakshiShreya) [inline Topological_Sort/Topological_Sort.js]: Add semicolon.

File: Topological_Sort/Topological_Sort.js (hunk 1 above)
Comment (JanaSabuj) [reply Topological_Sort/Topological_Sort.js]: @SakshiShreya this format helps in better visualization of the JS array destructuring.

**Comment by JanaSabuj** [comment]
@SakshiShreya  please review the changes.

File: Topological_Sort/Topological_Sort.js (hunk 3 above)
Comment (JanaSabuj) [reply Topological_Sort/Topological_Sort.js]: @SakshiShreya done.

**Comment by JanaSabuj** [comment]
Please review it. It has been very long. @SakshiShreya @jainaman224 @asha15

**Pull request merged**